indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -179,11 +192,20 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
    */
   public TaskLock lock(final Task task, final Interval interval) throws InterruptedException
   {
+    long startTime = System.currentTimeMillis();
     giant.lock();
     try {
       Optional<TaskLock> taskLock;
       while (!(taskLock = tryLock(task, interval)).isPresent()) {
-        lockReleaseCondition.await();
+        lockReleaseCondition.await(lockTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
A spurious wakeup can occur while awaiting (https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Condition.html). If it wakes up early but couldn't get a lock yet, the waiting time should be updated.